pkg/skaffold/runner/intent.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runner
+
+import gosync "sync"
+
+type intents struct {
+	build  bool
+	sync   bool
+	deploy bool
+
+	resetBuild  func()
+	resetSync   func()
+	resetDeploy func()
+
+	lock *gosync.Mutex
+}
+
+func NewIntents() *intents {
+	return &intents{
+		lock:        &gosync.Mutex{},
+		resetBuild:  func() {},
```

Comment:
1) these reset functions seem to be empty and never get anything interesting assigned to them, but they are called from all over the place. Am I missing something?:) 
2) if this is a genuine bug, we should cover them first with 3 tests and then implement

Reply:
I moved the logic for these functions into the constructor for the `intents` struct.